src/format/Kdbx3Writer.cpp

```diff
@@ -137,7 +137,7 @@ bool Kdbx3Writer::writeDatabase(QIODevice* device, Database* db)
         return false;
     }
 
-    KdbxXmlWriter xmlWriter(formatVersion());
+    KdbxXmlWriter xmlWriter(db->formatVersion());
```

Comment:
I guess this means in theory we could have KDBX 3.1 headers and KDBX 4.1 XML insides? Would KeePass2 barf on this combination? Would we barf on it on read?

Reply:
We use the same XML reader with a few version checks anyway. This has always been the case and would in theory be possible. I don't know about KeePass, but KeePassXC would be able to handle that case most likely. Reading a KDBX4.1 XML with a KDBX 4 header is definitely possible. If you try that across major versions, there might be errors when reading dates or other things that are hard incompatible (maybe, never tested it). But overall, we are pretty lenient and ignore stuff we don't know or wouldn't expect in that version anyway. The envelope version bytes are a mere hint.